Comptime error when query a field on array

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 21.10
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
CC version: cc (Ubuntu 11.2.0-7ubuntu2) 11.2.0

getwd: /var/www/vproject
vmodules: /home/tanguy/.vmodules
vroot: /var/www/vproject/v
vexe: /var/www/vproject/v/v
vexe mtime: 2022-05-03 06:38:58
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 4da2908.0e5c1ce

Git version: git version 2.32.0
Git vroot status: weekly.2022.16-130-g0e5c1cee
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 eee03626

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg /var/www/vproject/test/array_filter_struct_bug_copy.v`
```v
module main

pub struct Row {
pub mut:
	value string
}

fn main() {
	a := []Row{}
	for i in a {
		$for field in Row.fields {
			println('field $i: ' + a.$(field.name).str() )
		}
	}
}
```

**What did you expect to see?**

A formatted error explaining the a variable doesn't contain any field.


**What did you see instead?**
```
==================
/tmp/v_1000/array_filter_struct_bug_copy.12700846309302321918.tmp.c:11508: error: field not found: value
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -2660,7 +2660,7 @@ pub fn (mut c Checker) expr(node_ ast.Expr) ast.Type {
 			return c.comptime_call(mut node)
 		}
 		ast.ComptimeSelector {
-			node.left_type = c.unwrap_generic(c.expr(node.left))
+			node.left_type = c.expr(node.left)
 			expr_type := c.unwrap_generic(c.expr(node.field_expr))
 			expr_sym := c.table.sym(expr_type)
 			if expr_type != ast.string_type {

--- a/vlib/v/checker/tests/comptime_selector_expr_type_err.out
+++ b/vlib/v/checker/tests/comptime_selector_expr_type_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/comptime_selector_expr_type_err.vv:12:27: cgen error: `a` has no field named `value`
+   10 |     for i in a {
+   11 |         $for field in Row.fields {
+   12 |             println('field $i: ' + a.$(field.name).str() )
+      |                                    ^
+   13 |         }
+   14 |     }

--- a/vlib/v/checker/tests/comptime_selector_expr_type_err.vv
+++ b/vlib/v/checker/tests/comptime_selector_expr_type_err.vv
@@ -1,0 +1,15 @@
+module main
+
+pub struct Row {
+pub mut:
+	value string
+}
+
+fn main() {
+	a := []Row{}
+	for i in a {
+		$for field in Row.fields {
+			println('field $i: ' + a.$(field.name).str() )
+		}
+	}
+}

--- a/vlib/v/gen/c/comptime.v
+++ b/vlib/v/gen/c/comptime.v
@@ -20,7 +20,12 @@ fn (mut g Gen) comptime_selector(node ast.ComptimeSelector) {
 		if node.field_expr.expr is ast.Ident {
 			if node.field_expr.expr.name == g.comptime_for_field_var
 				&& node.field_expr.field_name == 'name' {
-				g.write(c_name(g.comptime_for_field_value.name))
+				field_name := g.comptime_for_field_value.name
+				left_sym := g.table.sym(g.unwrap_generic(node.left_type))
+				_ := g.table.find_field_with_embeds(left_sym, field_name) or {
+					g.error('`$node.left` has no field named `$field_name`', node.left.pos())
+				}
+				g.write(c_name(field_name))
 				return
 			}
 		}

